server/sonar-server/src/main/java/org/sonar/server/permission/ws/DeleteTemplateAction.java

```diff
@@ -75,6 +75,7 @@ private void doHandle(DeleteTemplateWsRequest request) {
       PermissionTemplateDto template = finder.getTemplate(dbSession, newTemplateRef(request.getTemplateId(), request.getTemplateName()));
       checkTemplateUuidIsNotDefault(template.getUuid());
       dbClient.permissionTemplateDao().deleteById(dbSession, template.getId());
+      dbClient.permissionTemplateCharacteristicDao().deleteByTemplateId(dbSession, template.getId());
```

Comment:
I think `dbClient.permissionTemplateDao().deleteById(…)` should call `permissionTemplateCharacteristicDao().deleteByTemplateId(…)` because it's already calling the delete methods for perm_templates_users and perm_templates_groups. We know responsibilities are not clearly separated, but it would be more coherent.